packages/@aws-cdk/aws-stepfunctions/lib/states/pass.ts

```diff
@@ -51,7 +51,7 @@ export interface PassProps {
      *
      * @default No injected result
      */
-    readonly result?: {[key: string]: any};
+    readonly result?: {[key: string]: any} | string | number | boolean | null | any[];
```

Comment:
We can't use type unions, at least not like this.

Is this actually blocking you somewhere?

Reply:
My previous state machine (0.33.0) was using `result: true`, which I had to modify to `result: {something: true}` when upgrading to 0.34.0.

My intuition was that if CFN supports a feature (in this case, `result` can be `{[key: string]: any} | string | number | boolean | null | any[]`), the CDK should allow it. Correct me if I'm wrong here :)

I can explore going down the "union-like class" path if you prefer.

Reply:
Yes, that will have to be the solution then.